src/Database/Connection.php

```diff
@@ -828,10 +829,10 @@ public function isQueryLoggingEnabled(): bool
     /**
      * Sets a logger
      *
-     * @param \Cake\Database\Log\QueryLogger|null $logger Logger object
+     * @param \Psr\Log\LoggerInterface $logger Logger object
      * @return $this
      */
-    public function setLogger(?QueryLogger $logger)
+    public function setLogger(LoggerInterface $logger)
```

Comment:
How does someone remove the logger now?

Reply:
In what scenario would removing a logger be required?

One should either not set a logger at all if he doesn't want logging or to prevent logging temporarily just disable logging.

Reply:
`setLogger()` internally creates of `QueryLogger` if `Connection::$_log` property is `null`. So using this method to unset the logger doesn't actually achieve anything currently either.

Reply:
could we also provide a `Null` logger?

Reply:
`psr/log` already provides a `NullLogger` :slightly_smiling_face:

Reply:
Making the argument non-nullable also allows to make the class implement `LoggerAwareInterface`.

Reply:
The we just need to add to the docs how using a null logger is possible. Even if it is obvious and redundant, I did not remember that `psr/log` had it.

Reply:
Passing a null logger works for me.